proxy/src/rate_limiter/limiter.rs

```diff
@@ -176,7 +188,7 @@ impl EndpointRateLimiter {
             self.map.len()
         );
         let n = self.map.shards().len();
-        let shard = thread_rng().gen_range(0..n);
```

Comment:
this lock is ok as the periodic cycle of `do_gc` makes this very unlikely to collide (impossible, infact, unless we have 2048 threads)